Mathlib/NumberTheory/Padics/PadicNorm.lean

```diff
@@ -305,6 +305,24 @@ theorem nat_lt_one_iff (m : ℕ) : padicNorm p m < 1 ↔ p ∣ m := by
   rw [← Int.coe_nat_dvd, ← int_lt_one_iff, Int.cast_ofNat]
 #align padic_norm.nat_lt_one_iff padicNorm.nat_lt_one_iff
 
+/-- If a rational is not a p-adic integer, it is not an integer. -/
+theorem not_int_of_not_padic_int (p : ℕ) {a : ℚ} [hp : Fact (Nat.Prime p)]
+    (H : 1 < padicNorm p a) : ¬ a.isInt := by
+  suffices : a.den ≠ 1; simpa [Rat.isInt]
+  by_cases h : a = 0
+  · subst h; contradiction
+  · unfold padicNorm at H
+    split_ifs at H; contradiction
+    suffices padicValRat p a < 0 by
+      rw [padicValRat_def] at this
+      intro Hden
+      rw [Hden] at this; simp only [padicValNat.one, sub_zero] at this
+      norm_cast
+    refine' neg_of_neg_pos _
+    have hx : (1 : ℚ) < p := by norm_cast; exact Nat.Prime.one_lt hp.elim
+    rw [← zpow_lt_iff_lt hx]
+    simpa only [zpow_zero]
```

Comment:
This feels simpler. That being said, I feel like `Rat.isInt` isn't quite the right tool for the job, since it is `Bool` and not `Prop`.
```suggestion
lemma _root_.Rat.isInt.eq_num {q : ℚ} (h : q.isInt) : q = q.num := by
  rw [Rat.isInt, Nat.beq_eq_true_eq] at h
  exact (Rat.coe_int_num_of_den_eq_one h).symm

/-- If a rational is not a p-adic integer, it is not an integer. -/
theorem not_int_of_not_padic_int (p : ℕ) {a : ℚ} [hp : Fact (Nat.Prime p)]
    (H : 1 < padicNorm p a) : ¬ a.isInt := by
  contrapose! H
  rw [Rat.isInt.eq_num H]
  apply padicNorm.of_int
```